### PR TITLE
Fix: withSigner to allow multiple signers within the same provider

### PR DESCRIPTION
**Pull request by adrianhdez929**
feat: add an indexOrAddress selector for multi signer provider

### CODE REVIEW

**Comment by adrianhdez929** [comment]
This PR to #70 is intended to not modify the legacy codebase while fixing the functionality, any change suggestion is appreciated, but I tried to modify existing code as possible since the current `withRunner` approach is not working

File: test/token/erc20.test.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    it("should transfer tokens between accounts using withRunner", async function () {
+        const preBalance = await wrapper.contractCall.balanceOf(await owner.getAddress())
+        const amount = parseEther("10")
+
+        await wrapper.withSigner(await alice.getAddress()).transfer(await owner.getAddress(), amount)
+        await wrapper.withSigner(await bob.getAddress()).transfer(await owner.getAddress(), amount)
+
+        const postBalance = await wrapper.contractCall.balanceOf(await owner.getAddress())
+
+        expect(postBalance).to.equal(preBalance+amount*2n);
```
Comment (MakeMake23) [inline test/token/erc20.test.ts:65]: Please, check the `transfer()` is decreasing the `alice` and `bob` balances.

Comment (MakeMake23) [reply test/token/erc20.test.ts:65]: Extend to other tests.

**Review by MakeMake23 — CHANGES_REQUESTED** [review]

**Comment by MakeMake23** [comment]
@adrianhdez929 please give me feedback on [my only comment](https://github.com/Fusyona/web3-core/pull/73#discussion_r1860919773).

**Comment by MakeMake23** [comment]
@TheSnakeWitcher please review this PR

**Comment by TheSnakeWitcher** [comment]
> @TheSnakeWitcher please review this PR

I did it already, I think it is ok but I personally prefer the approach in #70 because I think that it is unnecesarry to store in memory `signer` and `signerIndexOrAddress`, rationale is that `signer` is handled by the `contract` type that we wrap and `signerIndexOrAddress` it is responsability of the wallet

File: test/token/erc20.test.ts (hunk 1 above)
Comment (adrianhdez929) [reply test/token/erc20.test.ts:65]: I checked it at https://github.com/Fusyona/web3-core/pull/73/commits/14c09139208815cd9570d0763bb824a49bc1d2ec but what we want to achieve with that suite is to test the wrapper method calls, not the contract implementation itself, which is what that check does. Correct me if I'm wrong

**Comment by MakeMake23** [comment]
> > @TheSnakeWitcher please review this PR
> 
> I did it already, I think it is ok but I personally prefer the approach in #70 because I think that it is unnecesarry to store in memory `signer` and `signerIndexOrAddress`, rationale is that `signer` is handled by the `contract` type that we wrap and `signerIndexOrAddress` it is responsability of the wallet

@TheSnakeWitcher #70 was not working, that's why this PR exists.

**Comment by TheSnakeWitcher** [comment]
> > > @TheSnakeWitcher please review this PR
> > 
> > 
> > I did it already, I think it is ok but I personally prefer the approach in #70 because I think that it is unnecesarry to store in memory `signer` and `signerIndexOrAddress`, rationale is that `signer` is handled by the `contract` type that we wrap and `signerIndexOrAddress` it is responsability of the wallet
> 
> @TheSnakeWitcher #70 was not working, that's why this PR exists.

I fix it in [0426c2d](https://github.com/Fusyona/web3-core/pull/70/commits/0426c2de05b6fbf412e35bd0e4cdae1153494263), but anyways, since the investment in these PR is greater and the code is more refined we should move forward with these 👍🏿

**Pull request merged**